[ios] Fix errors thrown from SecureStore and other legacy modules

tsapeta opened the pull request:
# Why

It basically fixes two things:
- Empty `message` in errors thrown by SecureStore module
- Unspecified `code` (actually `"ERR_"`) in all errors thrown from the legacy modules when the native function is called by the JSI proxy

But also fixes invisible placeholders on SecureStore screen in NCL

# How

- Fixed `reject` calls in `EXSecureStore.m` by passing error messages as the second argument (it was incorrectly passed as the third argument)
- Made it possible to override error codes when initializing the base `Exception` class (it's used when calling legacy module methods through the JSI)
- Added `placeholderTextColor` prop to text inputs on the NCL screen

# Test Plan

- Disabled permission for FaceID for bare-expo
- Tried to get/set/delete keys from SecureStore with authentication option enabled
- Confirmed that returned errors contain correct codes and messages

![IMG_0953](https://user-images.githubusercontent.com/1714764/195866479-4179769c-4a8c-42ae-83fb-dd4974a0b953.jpg)

tsapeta committed:
Fix invisible placeholder on SecureStore screen in NCL

tsapeta committed:
Fix errors thrown from SecureStore and other legacy modules

brentvatne reviewed (approved):
it would be nice to have some tests for errorCodeFromString

tsapeta commented:
Yeah, I added this to my todo list. For now I extracted this only temporarily, eventually it will be a part of the `Exception` class and `CodedError` will be removed.
`Exception` already has a test spec, so I'll for sure update tests on that occasion.
